storage/connect/CMakeLists.txt

```diff
@@ -107,12 +110,22 @@ IF(CONNECT_WITH_LIBXML2)
         D:/libxml/lib)
   ENDIF(WIN32)
   FIND_PACKAGE(LibXml2)
+  SET_PACKAGE_PROPERTIES(LibXml2 PROPERTIES
+    DESCRIPTION "XML parsing library"
+    PURPOSE "Enables XML support in the CONNECT storage engine"
+    TYPE OPTIONAL
+  )
   IF (LIBXML2_FOUND)
     INCLUDE_DIRECTORIES(${LIBXML2_INCLUDE_DIR})
     SET(XML_LIBRARY  ${LIBXML2_LIBRARIES})
     SET(CONNECT_SOURCES ${CONNECT_SOURCES} libdoc.cpp libdoc.h)
     add_definitions(-DLIBXML2_SUPPORT)
+    add_feature_info(CONNECT_LIBXML2 "ON" "Support for LIBXML2 in the CONNECT storage engine")
+  ELSE(LIBXML2_FOUND)
+    add_feature_info(CONNECT_LIBXML2 "OFF" "Support for LIBXML2 in the CONNECT storage engine")
   ENDIF(LIBXML2_FOUND)
+ELSE(CONNECT_WITH_LIBXML2)
+  add_feature_info(CONNECT_LIBXML2 "OFF" "Support for LIBXML2 in the CONNECT storage engine")
```

Comment:
here again, can you please try to eliminate the else() branches added solely for the purpose of add_feature info.